pkg/kubectl/cmd/apply.go

```diff
@@ -240,6 +240,17 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 			visitedNamespaces.Insert(info.Namespace)
 		}
 
+		// Add change-cause annotation to resource info if it should be recorded
+		if cmdutil.ShouldRecord(cmd, info) {
+			recordInObj := info.VersionedObject
```

Comment:
NVM
It's fine. The `record` change is actually in `modified`. And it is handled correctly by function `GetModifiedConfiguration`.